zipline/algorithm.py

```diff
@@ -248,13 +244,10 @@ def __init__(self, *args, **kwargs):
 
         self._most_recent_data = None
 
-        # Subclasses that override initialize should only worry about
-        # setting self.initialized = True if AUTO_INITIALIZE is
-        # is manually set to False.
+        # Prepare the algo for initialization
         self.initialized = False
-        self.initialize(*args, **kwargs)
-        if self.AUTO_INITIALIZE:
-            self.initialized = True
+        self.initialize_args = args
```

Comment:
The initialize methods in AlgorithmProxy and LiveSessionAlgorithmProxy have different signatures. I will chalk it on to the forthcoming Zipline initialization cleanup/refactoring.